src/Widgets/Views/MainView.vala

```diff
@@ -26,9 +26,13 @@ public class Bluetooth.Widgets.MainView : Gtk.Box {
     private Gtk.Box devices_box;
     private Gtk.Revealer revealer;
 
+    private Gee.HashMap <string, Bluetooth.Widgets.Device> devices;
```

Comment:
What I think should be done here, is you should be using existing `get_devices` method inside the `ObjectManager` on demand and filter them manually, rather than creating a new HashMap.